### PR TITLE
feature/review

### PR DESCRIPTION
**Pull request by Kgomotso196**
# PROJECT TO DOs
### IN THIS PROJECT I:

1. Used webpack to bundle JavaScript.
2. Learnt how to use proper ES6 syntax.
3. Used ES6 modules to write modular JavaScript.
4. Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
5. Made sure that I used correct [GitHub Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
6. Made sure that I documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
7. Removed all hardcoded items from the tasks array.
8. Created a new JavaScript file for the new functionality.
9. Implemented a function for adding a new task (add a new element to the array).
10. Implemented a function for deleting a task (remove an element from the array).
11. Implemented a function for editing task descriptions.
12. Added event listener to the checkbox (change).
13. Updated items object's value for completed key upon user actions.
14. Implemented a function for the "Clear all completed" button (use filter() method).
15. Stored the updated array of items in local storage, so the user gets the correct list of values after the page reloads.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*eslint-disable*/
+import _ from 'lodash'; /*eslint-disable*/
```
Comment (cosywasswa) [inline src/index.js]: Kindly remove this as its an unused in this project.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  submitButton.onclick = function () {
+    Functionality.newTask();
+  };
+
+taskContainer.addEventListener('click', (e) => {
+  if (e.target !== null && e.target !== 'NaN' && e.target !== '') {
+    if (e.target.className === 'checkbox-class') {
+      const ids = e.target.id.replace('checkbox-', '');
+      let description = document.getElementById('d' + ids);
+      let data = Functionality.getAllTasks();
```
Comment (cosywasswa) [inline src/index.js]: Some Variable names are not descriptive enough to indicate their purpose, kindly consider that

**Review by cosywasswa — CHANGES_REQUESTED** [review]
Good job on implementing all the functionalities required. here are a few changes to work on regarding best practices.
happy coding.

File: src/index.js (hunk 1 above)
Comment (Kgomotso196) [reply src/index.js]: Hello @cosywasswa . Thank you for your kind insights. However, I used anonymous function in my case. When the submitButton is clicked, the anonymous function will be executed. In this case, it is assumed that the function Functionality.newTask() is called inside the anonymous function, indicating that it handles the creation of a new task. For that I got an error "unnamed function".

File: src/index.js (hunk 2 above)
Comment (Kgomotso196) [reply src/index.js]: Done.👍🏿

**Review by cosywasswa — APPROVED** [review]
Your project is complete and very impressive 💯! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉
approve2

**Pull request merged**